product_multi_barcode/models/product_barcode.py

```diff
@@ -0,0 +1,54 @@
+# © 2012-2014 Guewen Baconnier (Camptocamp SA)
+# © 2015 Roberto Lizana (Trey)
+# © 2016 Pedro M. Baeza
+# © 2018 Xavier Jimenez (QubiQ)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductBarcode(models.Model):
+    _name = "product.barcode"
+    _description = "Individual item in a product's barcode list"
+    _order = "sequence, id"
+
+    name = fields.Char(string="Barcode", required=True,)
+    sequence = fields.Integer(string="Sequence", default=0,)
+    product_id = fields.Many2one(
+        string="Product",
+        comodel_name="product.product",
+        compute="_compute_product",
+        store=True,
+        readonly=False,
+    )
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        compute="_compute_product_tmpl",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("product_id")
+    def _compute_product_tmpl(self):
+        for rec in self.filtered(lambda x: not x.product_tmpl_id and x.product_id):
+            rec.product_tmpl_id = rec.product_id.product_tmpl_id
+
+    @api.depends("product_tmpl_id.product_variant_ids")
+    def _compute_product(self):
+        for rec in self.filtered(
+            lambda x: not x.product_id and x.product_tmpl_id.product_variant_ids
+        ):
+            rec.product_id = rec.product_tmpl_id.product_variant_ids[0]
```

Comment:
```suggestion
    product_id = fields.Many2one(
        "product.product",
        string="Product",
        required=True,
    )
    product_tmpl_id = fields.Many2one(
        related="product_id.product_tmpl_id",
        store=True,
    )
```

What's the point of these computed fields?

I take it's to be able to write on `product_tmpl_id` (and update `product_id` accordingly), but what's the point? Barcodes should be linked to a `product.product` anyway.

Without this, creating barcodes from product.template wouldn't work as it is, but.. -> (see the other comment below)